lib/terminal/terminal_interpreter.dart

```diff
@@ -0,0 +1,482 @@
+      }
+    }
+
+    // any remaining unprocessed number
+    if (num > 0) res.add((num / 10).toInt());
+
+    return res;
+  }
+
+  void _eraseFromCursorToEnd() {
+    for (int i = currentColumn; i < columns; i++) {
+      bufferState[currentLine][i].character = ' ';
+      bufferState[currentLine][i].backgroundColor = _backgroundColor;
+      bufferState[currentLine][i].foregroundColor = _foregroundColor;
+    }
+  }
+
+  void _eraseFromBeginningToCursor() {
+    for (int i = 0; i <= currentColumn; i++) {
+      bufferState[currentLine][i].character = ' ';
+      bufferState[currentLine][i].backgroundColor = _backgroundColor;
+      bufferState[currentLine][i].foregroundColor = _foregroundColor;
+    }
+  }
+
+  void _eraseEntireLine() {
+    for (int i = 0; i < columns; i++) {
+      bufferState[currentLine][i].character = ' ';
+      bufferState[currentLine][i].backgroundColor = _backgroundColor;
+      bufferState[currentLine][i].foregroundColor = _foregroundColor;
+    }
+  }
+
+  String _matrixToString(List<List<String>> matrix) {
+    return matrix.map((row) => row.join()).join('\n');
+  }
+
+  String _formatMatrixToString(List<List<String>> matrix) {
+    final padded = _padMatrix(matrix);
+    return padded.map((row) => row.join()).join('\n');
+  }
+
+  List<List<String>> _padMatrix(List<List<String>> matrix) {
+    return matrix
+        .map((row) => row.map((cell) => cell.padRight(8)).toList())
+        .toList();
+  }
+
+  bool _canConvertToNumber(String char) {
+    return int.tryParse(char) != null;
+  }
+
+  void _clearStyles() {
+    _foregroundColor = null;
+    _backgroundColor = null;
+    _fontStyles.clear();
+  }
+
+  bool _validRgbColor(int r, int g, int b) {
+    bool validRed = 0 <= r && r < 256;
+    bool validGreen = 0 <= g && g < 256;
+    bool validBlue = 0 <= b && b < 256;
+
+    return validRed && validGreen && validBlue;
+  }
+
+  AnsiColorType _colorFromCode(int code) {
+    final mapping = <int, Colors>{
+      30: Colors.black,
+      31: Colors.red,
+      32: Colors.green,
+      33: Colors.yellow,
+      34: Colors.blue,
+      35: Colors.magenta,
+      36: Colors.cyan,
+      37: Colors.white,
+      90: Colors.highBlack,
+      91: Colors.highRed,
+      92: Colors.highGreen,
+      93: Colors.highYellow,
+      94: Colors.highBlue,
+      95: Colors.highMagenta,
+      96: Colors.highCyan,
+      97: Colors.highWhite,
+      40: Colors.black,
+      41: Colors.red,
+      42: Colors.green,
+      43: Colors.yellow,
+      44: Colors.blue,
+      45: Colors.magenta,
+      46: Colors.cyan,
+      47: Colors.white,
+      100: Colors.highBlack,
+      101: Colors.highRed,
+      102: Colors.highGreen,
+      103: Colors.highYellow,
+      104: Colors.highBlue,
+      105: Colors.highMagenta,
+      106: Colors.highCyan,
+      107: Colors.highWhite,
+    };
+    return mapping[code]!;
+  }
+
+  FontStyle _styleFromCode(int code) {
+    final mapping = <int, FontStyle>{
```

Comment:
Ditto, can this be a const field?

Also probably make it a `FontStyle.fromCode(int code)` factory constructor.